app/vite.config.ts

```diff
@@ -1,25 +1,62 @@
 /// <reference types="vitest" />
 /// <reference types="vite/client" />
 
-import {defineConfig} from 'vite';
+import {defineConfig, loadEnv} from 'vite';
 import react from '@vitejs/plugin-react';
 
+// Returns the value of the VITE_HUU_API_BASE_URL environment variable as a URL.
+// This ensure the environment variable exists and the value is a URL.
+function huuApiBaseUrl(envHuuApiBaseUrl: string, mode: string): URL | never {
+  try {
+    return new URL(envHuuApiBaseUrl);
+  } catch {
+    if (mode == 'development' || mode == 'test') {
+      return new URL('http://localhost:4040/api/');
+    } else {
+      throw new Error('VITE_HUU_API_BASE_URL is not configured with a URL');
+    }
+  }
+}
+
+// This code configures vite before it processes the application's code
 // https://vitejs.dev/config/
-export default defineConfig({
-  plugins: [react()],
-  server: {
-    port: 4040,
-    proxy: {
-      '/api': {
-        target: 'http://localhost:8080',
-        changeOrigin: true,
-        secure: false,
+export default defineConfig(({mode}) => {
+  // Load environment variables starting with VITE_ from:
+  // 1. the process' environment variables
+  // 2. if exists, a .env file and any .env files for the given mode
+  //
+  // The process' environment variables take precedence over variables defined in any .env file.
+  //
+  // The environment variables can be used in code using `import.meta.env.VITE_<rest of variable>`.
+  // They are statically string replaced during a production build
+  //
+  // By default, vite automatically loads the environment variables.
+  // However, they are explicitly loaded here to handle the required variables.
+  // Otherwise, vite will replace undefined environment variables with `{}.VITE_<rest of variable>`.
+  const env = loadEnv(mode, process.cwd());
+
+  return {
+    define: {
+      // ensures this is defined for vite to replace
+      'import.meta.env.VITE_HUU_API_BASE_URL': JSON.stringify(
+        huuApiBaseUrl(env.VITE_HUU_API_BASE_URL, mode),
+      ),
+    },
+    plugins: [react()],
+    server: {
+      port: 4040,
+      proxy: {
+        '/api': {
+          target: 'http://localhost:8080',
+          changeOrigin: true,
+          secure: false,
```

Comment:
This is a development-only server proxy, right? Just want to make sure that allowing unsecured proxies does not pose a security risk to the production application. 

I believe we use the VITE server for development, and `nginx` for production right?

Reply:
@Joshua-Douglas yes development only. The production front-end is compiled into html, JS, css files and served by Nginx.

Reply:
A static site.